modules/tinymce/src/core/test/ts/browser/util/ImageUploaderTest.ts

```diff
@@ -0,0 +1,57 @@
+import { Log, Pipeline, Chain } from '@ephox/agar';
+import { Assert, UnitTest } from '@ephox/bedrock-client';
+import { ApiChains, TinyLoader } from '@ephox/mcagar';
+import { Arr } from '@ephox/katamari';
+import ImageUploader, { UploadResult } from 'tinymce/core/api/util/ImageUploader';
+
+import { BlobCache, BlobInfo } from 'tinymce/core/api/file/BlobCache';
+
+import Theme from 'tinymce/themes/silver/Theme';
+
+UnitTest.asynctest('browser.tinymce.core.util.ImageUploaderTest', (success, failure) => {
+
+  Theme();
+
+  TinyLoader.setupLight((editor, onSuccess, onFailure) => {
+    const cache = BlobCache();
+    const blob = new Blob([ JSON.stringify({ hello: 'world' }) ]);
+
+    const image1 = cache.create({ blob, base64: 'test' });
+
+    const cUploadImages = (images: BlobInfo[], notifications: boolean) => Chain.async((_value, next, die) => {
+      const uploader = ImageUploader(editor);
+      uploader.upload(images, notifications).then(next, die);
+    });
+
+    const cAssertUploadResultSuccess = (expectedUrl: string, success: boolean) => Chain.op((uploadResults: UploadResult[]) => {
+      Arr.each(uploadResults, (uploadResult) => {
+        Assert.eq('Url is Image.png', expectedUrl, uploadResult.url);
+        Assert.eq('Upload result status is true upon success', success, uploadResult.status);
+      });
+    });
+
+    const cAssertUploadResultFailure = (expectedUrl: string, error: boolean) => Chain.op((uploadResults: UploadResult[]) => {
```

Comment:
Once again I don't think we want to be passing a bool here. This should be the error message string that we should assert further down.

```suggestion
    const cAssertUploadResultFailure = (expectedUrl: string, errorMsg: string) => Chain.op((uploadResults: UploadResult[]) => {
```

Reply:
error return is a bool, not a string :-)

Reply:
Yes, this is in reference to [this comment](https://github.com/tinymce/tinymce/pull/6203#discussion_r522543887), I think we should have the status hardcoded in the tests because this test chain is specifically to assert a failure.

Instead, we should pass through the expected error message and assert that `uploadResult.error.message` matches the message that was provided when the test calls `failure('Error message')`

Reply:
Done